pkg/azure/actuator.go

```diff
@@ -230,36 +241,53 @@ func (a *Actuator) sync(ctx context.Context, cr *minterv1.CredentialsRequest) er
 		logger.Debug("credentials already up to date")
 		return nil
 	}
-
-	credentialsRootSecret, err := a.GetCredentialsRootSecret(ctx, cr)
+	stsDetected, err := utils.IsTimedTokenCluster(a.client, ctx, logger)
 	if err != nil {
-		logger.WithError(err).Error("issue with cloud credentials secret")
 		return err
 	}
-
-	switch credentialsRootSecret.Annotations[constants.AnnotationKey] {
-	case constants.InsufficientAnnotation:
-		msg := "cloud credentials insufficient to satisfy credentials request"
-		logger.Error(msg)
-		return &actuatoriface.ActuatorError{
-			ErrReason: minterv1.InsufficientCloudCredentials,
-			Message:   msg,
+	if stsDetected {
+		logger.Debug("actuator detected Azure AD Workload Identity enabled cluster, enabling Workload Identity secret brokering for CredentialsRequests providing a Managed Identity")
+		azureProviderSpec, err := decodeProviderSpec(minterv1.Codec, cr)
+		if err != nil {
+			return err
 		}
-	case constants.PassthroughAnnotation:
-		logger.Debugf("provisioning with passthrough")
-		err := a.syncPassthrough(ctx, cr, credentialsRootSecret, logger)
+		azureFederatedTokenFile := cr.Spec.CloudTokenPath
+		if cr.Spec.CloudTokenPath == "" {
+			logger.Debug("CredentialsRequest has no cloudTokenPath, defaulting azure_federated_token_file to /var/run/secrets/kubernetes.io/serviceaccount/token")
+			azureFederatedTokenFile = "/var/run/secrets/kubernetes.io/serviceaccount/token"
```

Comment:
Do we want to match the default to what is used by the `ccoctl` command? 
/var/run/secrets/openshift/serviceaccount/token

https://github.com/openshift/cloud-credential-operator/blob/3c72c02d745112f07d9ccd3675ac017e864a2c76/pkg/cmd/provisioning/constants.go#L17